packages/orchid/src/task.types.ts

```diff
@@ -1,56 +1,66 @@
-declare global {
-  // eslint-disable-next-line @typescript-eslint/no-empty-interface
-  interface GlobalTasks {}
+export type TaskSpec<
+  Input,
+  Output,
+  Context extends TaskContext = TaskContext
+> = TaskSpecObject<Input, Output, Context> | RunFn<Input, Output, Context>;
+
+type RunFn<Input, Output, Context = TaskContext> = (
+  input: Input,
+  ctx: Context
+) => Output;
+export interface TaskSpecObject<
+  Input,
+  Output,
+  Context extends TaskContext = TaskContext
+> {
+  id: string;
+  run(input: Input, ctx: Context): Output;
 }
 
-export type TaskSpec<
-  Input = unknown,
-  Output = unknown,
-  ParentInputType = null
-> = {
-  name: string;
-  run(input: Input, context: TaskContext<ParentInputType>): Output;
+export const wrapToObject = <Input, Output>(
+  task: TaskSpec<Input, Output>
+): TaskSpecObject<Input, Output> => {
+  if (typeof task === 'function') {
+    return {
+      id: task.name,
```

Comment:
Hmmm the more I think about it I find it tricky to call this `id` when we use the function name for it. E.g. when someone does this

```js
run(() => console.log('Where is your god now?'))
```

`task.name` will be an empty string because the function is anonymous. I think in that case it's better to give it the id `'anonymous'` or something like that. But that ID won't be unique then. Maybe `name` is better after all.

Reply:
You mean just as identifier? I think it makes sense, also since there is no real guarantee that the `id` is unique. On the other hand `id` is shorter than `name`. I think ill give it some thought and might change it tomorrow.

Reply:
Actually I gave it some thought and name is more representative of what it is.
0ae2321488d9488327256022665f5d02e2cad559

Reply:
Don't forget to give the task a default name in case it's an anonymous function!